pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -514,6 +514,12 @@ type CommonPrometheusFields struct {
 	// It is meant to be used by admins to keep the overall number of
 	// samples/series under a desired limit.
 	//
+	// When both `enforcedSampleLimit` and `sampleLimit` are defined and greater than zero, the following rules apply:
```

Comment:
shall we say that when sampleLimit > enforcedSampleLimit, we consider only enforcedSampleLimit?

Reply:
its already there I think.

```If Prometheus version is >= 2.45.0 and the `enforcedTargetLimit` is greater than the `targetLimit`, the `targetLimit` will be set to `enforcedTargetLimit`.```

Reply:
:+1: